facebook/delphiFacebook/R/contingency_variables.R

```diff
@@ -329,82 +359,738 @@ remap_responses <- function(df) {
+  } else {
+    df$hispanic <- NA
+  }
+
+  # raceethnicity
+  if ( "hispanic" %in% names(df) &
+       "race" %in% names(df) ) {
+    df$raceethnicity <- case_when(
+      df$hispanic ~ "Hispanic",
+      !df$hispanic & df$race == "AmericanIndianAlaskaNative" ~ "NonHispanicAmericanIndianAlaskaNative",
+      !df$hispanic & df$race == "Asian" ~ "NonHispanicAsian",
+      !df$hispanic & df$race == "BlackAfricanAmerican" ~ "NonHispanicBlackAfricanAmerican",
+      !df$hispanic & df$race == "NativeHawaiianPacificIslander" ~ "NonHispanicNativeHawaiianPacificIslander",
+      !df$hispanic & df$race == "White" ~ "NonHispanicWhite",
+      !df$hispanic & df$race == "Other" ~ "NonHispanicMultipleOther",
+      TRUE ~ NA_character_
+    )
+  } else {
+    df$raceethnicity <- NA_character_
+  }
+
+  # healthcareworker
+  if ("mc_occupational_group" %in% names(df)) {
+    df$healthcareworker <-
+      df$mc_occupational_group == "Healthcare support" |
+      df$mc_occupational_group == "Healthcare practitioner"
+  } else {
+    df$healthcareworker <- NA
+  }
+
+  # pregnant
+  if (all(c("mc_pregnant", "gender") %in% names(df))) {
+    df$pregnant <- case_when(
+      df$mc_pregnant == 1 ~ TRUE,
+      df$mc_pregnant == 2 ~ FALSE,
+      df$mc_pregnant == 4 ~ NA,
+      df$gender == "Male" ~ NA,
+      TRUE ~ NA
+    )
+  } else {
+    df$pregnant <- NA
+  }
+
+  # smoker
+  if ("D11" %in% names(df)) {
+    df$smoker <- as.numeric(df$D11 == 1)
+  } else {
+    df$smoker <- NA
+  }
+
+  # eligible
+  # comorbidheartdisease
+  # comorbidcancer
+  # comorbidkidneydisease
+  # comorbidlungdisease
+  # comorbiddiabetes
+  # comorbidimmuno
+  # comorbidobese
+  if ("ms_comorbidities" %in% names(df)) {
+    comorbidities <- split_options(df$ms_comorbidities)
+
+    df$comorbidheartdisease <- is_selected(comorbidities, "Heart disease")
+    df$comorbidcancer <- is_selected(comorbidities, "Cancer")
+    df$comorbidkidneydisease <- is_selected(comorbidities, "Kidney disease")
+    df$comorbidlungdisease <- is_selected(comorbidities, "Chronic lung disease")
+    df$comorbiddiabetes <-
+      is_selected(comorbidities, "Diabetes") |
+      is_selected(comorbidities, "Type 1 diabetes") |
+      is_selected(comorbidities, "Type 2 diabetes")
+    df$comorbidimmuno <- is_selected(comorbidities, "Compromised immune system")
+    df$comorbidobese <- is_selected(comorbidities, "Obesity")
+    df$eligible <- 
+      df$comorbidheartdisease |
+      df$comorbidcancer |
+      df$comorbidkidneydisease |
+      df$comorbidlungdisease |
+      df$comorbiddiabetes |
+      df$comorbidimmuno
+      
+  } else {
+    df$comorbidheartdisease <- NA
+    df$comorbidcancer <- NA
+    df$comorbidkidneydisease <- NA
+    df$comorbidlungdisease <- NA
+    df$comorbiddiabetes <- NA
+    df$comorbidimmuno <- NA
+    df$comorbidobese <- NA
+    df$eligible <- NA
+  }
+
+  # eligiblepregsmokeobese
+  if (all(c("eligible", "pregnant", "smoker", "comorbidobese") %in% names(df))) {
```

Comment:
This section depends on gender as well, so we should make sure it's in the names.

Reply:
Done.